File: app/src/main/java/com/example/shibeapplication/viewmodels/ShibeViewModel.java
package com.example.shibeapplication.viewmodels;

import com.example.shibeapplication.repositories.Repository;

import java.util.List;

import retrofit2.Call;

public class ShibeViewModel {

    public ShibeViewModel() {

    }

    public Call<List<String>> getShibes(int count) {
        return Repository.getInstance().getShibes(count);
    }

}

File: app/src/main/java/com/example/shibeapplication/viewmodels/ShibeViewModel.kt
package com.example.shibeapplication.viewmodels

import androidx.lifecycle.ViewModel
import com.example.shibeapplication.model.repositories.Repository
import dagger.hilt.android.lifecycle.HiltViewModel
import retrofit2.Call
import javax.inject.Inject

@HiltViewModel
class ShibeViewModel @Inject constructor(private val repository: Repository) : ViewModel() {

    fun getShibes(count: Int): Call<List<String>> {
        return repository.getShibes(count)
    }
}